core/src/main/java/org/apache/carbondata/core/metadata/datatype/ArrayType.java

```diff
@@ -30,4 +30,13 @@
   public boolean isComplexType() {
     return true;
   }
+
+  @Override
+  public int getNumOfChild() {
+    return 1;
```

Comment:
isn't it elementype.getNumOfChild

Reply:
I checked master code, it seems like this only